File: 0650-2-keys-keyboard/0650-2-keys-keyboard.py
class Solution:
    def minSteps(self, n: int) -> int:
        result = [0 for i in range(n + 1)]

        for i in range(2, n + 1):
            result[i] = i
            for j in range(i // 2, 1, -1):
                if i % j == 0:
                    result[i] = result[j] + i // j
                    break

        return result[-1]

File: 0650-2-keys-keyboard/0650-2-keys-keyboard.cpp
class Solution {
public:
    int minSteps(int n) {
        int result[n + 1];
        result[1] = 0;

        for (int i = 2; i < n + 1; i++) {
            result[i] = i;
            for (int j = i / 2; j > 1; j--) {
                if (i % j == 0) {
                    result[i] = result[j] + i / j;
                    break;
                }
            }
        }

        return result[n];
    }
};